java-lib/src/test/java/org/clink/feature/ClinkOneHotEncoderTest.java

```diff
@@ -0,0 +1,112 @@
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+    }
+
+    @Test
+    public void testFitAndPredict() {
+        ClinkOneHotEncoder estimator = new ClinkOneHotEncoder();
+        estimator.setInputCols("a", "b");
+        estimator.setOutputCols("c", "d");
+        estimator.setDropLast(false);
+
+        DataStream<Row> trainStream = env.fromElements(Row.of(0, 1), Row.of(2, 3));
+        Table trainTable = tEnv.fromDataStream(trainStream).as("a", "b");
+        DataStream<Row> predictStream = env.fromElements(Row.of(0, 1));
+        Table predictTable = tEnv.fromDataStream(predictStream).as("a", "b");
+
+        ClinkOneHotEncoderModel model = estimator.fit(trainTable);
+
+        Table outputTable = model.transform(predictTable)[0];
+
+        Row actual = outputTable.execute().collect().next();
+        Row expected =
+                Row.of(
+                        0,
+                        1,
+                        Vectors.sparse(2, new int[] {0}, new double[] {1.0}),
+                        Vectors.sparse(3, new int[] {1}, new double[] {1.0}));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testSaveAndLoad() throws Exception {
+        ClinkOneHotEncoder estimator = new ClinkOneHotEncoder();
+        estimator.setInputCols("a", "b");
+        estimator.setOutputCols("c", "d");
+        estimator.setDropLast(false);
+
+        DataStream<Row> trainStream = env.fromElements(Row.of(0, 1), Row.of(2, 3));
+        Table trainTable = tEnv.fromDataStream(trainStream).as("a", "b");
+        DataStream<Row> predictStream = env.fromElements(Row.of(0, 1));
+        Table predictTable = tEnv.fromDataStream(predictStream).as("a", "b");
+
+        ClinkOneHotEncoderModel modelA = estimator.fit(trainTable);
+
+        String savePath = tempFolder.newFolder().getAbsolutePath();
+        modelA.save(savePath);
+        env.execute();
+        ClinkOneHotEncoderModel modelB = ClinkOneHotEncoderModel.load(env, savePath);
+
+        Table outputTable = modelB.transform(predictTable)[0];
+
+        Row actual = outputTable.execute().collect().next();
+        Row expected =
+                Row.of(
+                        0,
+                        1,
+                        Vectors.sparse(2, new int[] {0}, new double[] {1.0}),
+                        Vectors.sparse(3, new int[] {1}, new double[] {1.0}));
+        assertEquals(expected, actual);
+    }
```

Comment:
Could we also have `testGetModelData()` and `testSetModelData()` similar to the existing Flink ML's OneHotEncoderTest?

Reply:
OK. I'll add them.